hardware/hardware.mk

```diff
@@ -68,15 +68,17 @@ system.v: $(SRC_DIR)/system_core.v
 PYTHON_DIR=$(MEM_DIR)/software/python
 
 boot.hex: $(BOOT_DIR)/boot.bin
-	$(PYTHON_DIR)/makehex.py $(BOOT_DIR)/boot.bin $(BOOTROM_ADDR_W) > boot.hex
+	$(PYTHON_DIR)/makehex.py $< $(BOOTROM_ADDR_W) > $@
 
 firmware.hex: $(FIRM_DIR)/firmware.bin
-	$(PYTHON_DIR)/makehex.py $(FIRM_DIR)/firmware.bin $(FIRM_ADDR_W) > firmware.hex
+	$(PYTHON_DIR)/makehex.py $< $(FIRM_ADDR_W) > $@
 	$(PYTHON_DIR)/hex_split.py firmware .
-	cp $(FIRM_DIR)/firmware.bin .
+
+firmware.bin: $(FIRM_DIR)/firmware.bin
```

Comment:
delete this target

Reply:
For FPGAs, the firmware.hex is no dependency when INIT_MEM=0. That is the problem that I solved on this pull-request.